private-templates-service/client/src/store/currentTemplate/types.ts

```diff
@@ -1,10 +1,13 @@
+import { Template } from 'adaptive-templating-service-typescript-node';
+
 import { IncomingMessage } from "http";
 
 export interface CurrentTemplateState {
-  templateID?: string
-  templateJSON?: string
-  templateName?: string
-  sampleDataJSON?: string
+  templateID?: string;
+  templateJSON?: string;
+  templateName?: string;
+  sampleDataJSON?: string;
+  template?: Template;
```

Comment:
Instead of having a separate templateID, templateJSON, templateName and sampleDataJSON field in state, now that we have a Template type state here can we change the store so that this state is only concerned with `template`?

Just a suggestion.

Reply:
Can we do that in a separate pr? I think that is not a trivial change.

Reply:
Sure.